src/gboml/scope.py

```diff
@@ -33,9 +34,27 @@ def _add_to_scope(self, ast, wrapper=lambda x: x, whenPresent: OverrideBehavior
             else:
                 pass
 
+        if ast.name in ['T', 't']:
+            raise KeyError(f"Identifier {ast.name} cannot be redifined (reserved keyword)")
+
         self.content[ast.name] = wrapper(create_scope(ast, self))
         return self.content[ast.name]
 
+    def _add_unnamed_to_scope(self, name, whenPresent: OverrideBehavior = OverrideBehavior.fail) -> None:
+        if name in self.content:
```

Comment:
I do not understand what this function does; can you document it?

Reply:
Basically, it will add names to the Scope. Those names are not `GBOMLObject`. They can be found e.g. in the `Loop`s with `loop.varid`, or in the `FunctionDefinition`s with `fct.args[]`.